server/app/test_pipeline.go

```diff
@@ -0,0 +1,106 @@
+package app
+
+import (
+	"github.com/kubeshop/tracetest/server/datastore"
+	"github.com/kubeshop/tracetest/server/executor"
+	"github.com/kubeshop/tracetest/server/executor/pollingprofile"
+	"github.com/kubeshop/tracetest/server/executor/testrunner"
+	"github.com/kubeshop/tracetest/server/executor/trigger"
+	"github.com/kubeshop/tracetest/server/linter/analyzer"
+	"github.com/kubeshop/tracetest/server/model"
+	"github.com/kubeshop/tracetest/server/subscription"
+	"github.com/kubeshop/tracetest/server/test"
+	"github.com/kubeshop/tracetest/server/tracedb"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func buildTestPipeline(
+	ppRepo *pollingprofile.Repository,
+	dsRepo *datastore.Repository,
+	lintRepo *analyzer.Repository,
+	trRepo *testrunner.Repository,
+	treRepo model.TestRunEventRepository,
+	testRepo test.Repository,
+	runRepo test.RunRepository,
+	tracer trace.Tracer,
+	subscriptionManager *subscription.Manager,
+	triggerRegistry *trigger.Registry,
+) *executor.TestPipeline {
+	eventEmitter := executor.NewEventEmitter(treRepo, subscriptionManager)
+
+	execTestUpdater := (executor.CompositeUpdater{}).
+		Add(executor.NewDBUpdater(runRepo)).
+		Add(executor.NewSubscriptionUpdater(subscriptionManager))
+
+	assertionRunner := executor.NewAssertionRunner(
+		execTestUpdater,
+		executor.NewAssertionExecutor(tracer),
+		executor.InstrumentedOutputProcessor(tracer),
+		subscriptionManager,
+		eventEmitter,
+	)
+
+	linterRunner := executor.NewlinterRunner(
+		execTestUpdater,
+		subscriptionManager,
+		eventEmitter,
+		lintRepo,
+	)
+
+	pollerExecutor := executor.NewSelectorBasedPoller(
+		executor.NewPollerExecutor(
+			tracer,
+			execTestUpdater,
+			tracedb.Factory(runRepo),
+			dsRepo,
+			eventEmitter,
+		),
+		eventEmitter,
+	)
+
+	tracePoller := executor.NewTracePoller(
+		pollerExecutor,
+		execTestUpdater,
+		subscriptionManager,
+		eventEmitter,
+	)
+
+	runner := executor.NewPersistentRunner(
+		triggerRegistry,
+		execTestUpdater,
+		tracer,
+		subscriptionManager,
+		tracedb.Factory(runRepo),
+		dsRepo,
+		eventEmitter,
+	)
+
+	cancelRunHandlerFn := executor.HandleRunCancelation(execTestUpdater, tracer, eventEmitter)
+
+	queueBuilder := executor.NewQueueBuilder().
+		WithCancelRunHandlerFn(cancelRunHandlerFn).
+		WithSubscriptor(subscriptionManager).
+		WithDataStoreGetter(dsRepo).
+		WithPollingProfileGetter(ppRepo).
+		WithTestGetter(testRepo).
+		WithRunGetter(runRepo)
+
+	pipeline := executor.NewPipeline(queueBuilder,
+		executor.PipelineStep{Processor: runner, Driver: executor.NewInMemoryQueueDriver("runner")},
+		executor.PipelineStep{Processor: tracePoller, Driver: executor.NewInMemoryQueueDriver("tracePoller")},
+		executor.PipelineStep{Processor: linterRunner, Driver: executor.NewInMemoryQueueDriver("linterRunner")},
```

Comment:
One thing to consider in the future: since we are considering renaming the internals of `environment` and `transaction` soon, we can also add the `linter` to the packet and change it to `analyzer` explicitly. Does it make sense?

Reply:
I agree, but I saw that there's a distinction between `linter` and `analyzer`. I didn't dig too much into it, but that looks like a good time to normalize this kinds of things